File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports DevExpress.XtraLayout.Registrator
Imports DevExpress.XtraLayout
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraLayout.Painting
Imports DevExpress.XtraLayout.ViewInfo
Imports DevExpress.Utils.Drawing
Imports DevExpress.Skins
Imports DevExpress.Utils
Imports System.Collections

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub myLayoutControl1_CustomDrawBackground(ByVal sender As Object, ByVal e As CustomDrawEventArgs) Handles myLayoutControl1.CustomDrawBackground
			If e.Args.Owner Is layoutControlGroup2 Then
				Dim bounds As Rectangle = (CType(e.Args.BorderInfo, GroupObjectInfoArgs)).ClientBounds
				e.Args.Graphics.FillRectangle(Brushes.Red, bounds)
				e.Args.Graphics.DrawString("Custom draw", AppearanceObject.DefaultFont, Brushes.Yellow, bounds)
				e.Handled = True
			End If
		End Sub

		Private Sub myLayoutControl1_CustomDrawControlsArea(ByVal sender As Object, ByVal e As CustomDrawEventArgs) Handles myLayoutControl1.CustomDrawControlsArea
			If e.Args.Owner Is layoutControlGroup3 Then
				For Each item As BaseLayoutItem In New ArrayList(layoutControlGroup3.Items)
					e.Args.Graphics.DrawString("--Custom draw--", AppearanceObject.DefaultFont, Brushes.Red, item.ViewInfo.BoundsRelativeToControl)
				Next item
				e.Handled = True
			End If
		End Sub


		Private Sub myLayoutControl1_CustomDrawSelection(ByVal sender As Object, ByVal e As CustomDrawEventArgs) Handles myLayoutControl1.CustomDrawSelection
			Dim group As LayoutControlGroup = TryCast(e.Args.Owner, LayoutControlGroup)
			If group IsNot Nothing Then
				If group.ViewInfo.State = ObjectState.Selected Then
					Dim rect As Rectangle = e.Args.PainterBoundsRelativeToControl
					rect.Inflate(-6, -6)
					e.Args.Graphics.DrawRectangle(New Pen(Brushes.Red, 5), rect)
				End If
			End If
			e.Handled = True

		End Sub

		Private Sub myLayoutControl1_CustomDrawTextArea(ByVal sender As Object, ByVal e As CustomDrawEventArgs) Handles myLayoutControl1.CustomDrawTextArea
			e.Handled = True
		End Sub
	End Class
End Namespace

File: CS/C#/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using DevExpress.XtraLayout.Registrator;
using DevExpress.XtraLayout;
using DevExpress.LookAndFeel;
using DevExpress.XtraLayout.Painting;
using DevExpress.XtraLayout.ViewInfo;
using DevExpress.Utils.Drawing;
using DevExpress.Skins;
using DevExpress.Utils;
using System.Collections;

namespace WindowsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void myLayoutControl1_CustomDrawBackground(object sender, CustomDrawEventArgs e)
        {
            if (e.Args.Owner == layoutControlGroup2)
            {
                Rectangle bounds = ((GroupObjectInfoArgs)(e.Args.BorderInfo)).ClientBounds;
                e.Args.Graphics.FillRectangle(Brushes.Red, bounds);
                e.Args.Graphics.DrawString("Custom draw", AppearanceObject.DefaultFont, Brushes.Yellow, bounds);
                e.Handled = true;
            }
        }

        private void myLayoutControl1_CustomDrawControlsArea(object sender, CustomDrawEventArgs e)
        {
            if (e.Args.Owner == layoutControlGroup3)
            {
                foreach (BaseLayoutItem item in new ArrayList(layoutControlGroup3.Items))
                    e.Args.Graphics.DrawString("--Custom draw--", AppearanceObject.DefaultFont, Brushes.Red, item.ViewInfo.BoundsRelativeToControl);
                e.Handled = true;
            }
        }


        private void myLayoutControl1_CustomDrawSelection(object sender, CustomDrawEventArgs e)
        {
            LayoutControlGroup group = e.Args.Owner as LayoutControlGroup;
            if (group != null)
                if (group.ViewInfo.State == ObjectState.Selected)
                {
                    Rectangle rect = e.Args.PainterBoundsRelativeToControl;
                    rect.Inflate(-6, -6);
                    e.Args.Graphics.DrawRectangle(new Pen(Brushes.Red, 5), rect);
                }
            e.Handled = true;

        }

        private void myLayoutControl1_CustomDrawTextArea(object sender, CustomDrawEventArgs e)
        {
            e.Handled = true;
        }
    }
}